PrimePython/solution_1/PrimePY.py

```diff
@@ -1,140 +1,125 @@
-
 # Python Prime Sieve
 #
 # MyFirstPython Program (tm) Dave Plummer 8/9/2018
 #
-# This is the main prime_sieve class. Call it with the number you wish as an upper limit, then
-# call the runSieve method to do the calculation. printResults will dump the count to check validity.
+# This is the main PrimeSieve class. Call it with the number you wish as an upper limit, then
+# call the run_sieve method to do the calculation. print_results will dump the count to check validity.
 #
 # Updated 3/22/2021 for Dave's Garage episode comparing C++, C#, and Python
-
-from sys import stdout                      # So I can print without an automatic python newline
-from math import sqrt                       # Used by the sieve
-import timeit                               # For timing the durations
-
-class prime_sieve(object):
-
-    rawbits = None   # Storage for sieve - since we filter evens, just half as many bits
-    sieveSize = 0    # Upper limit, highest prime we'll consider
-
-    primeCounts = { 10 : 4,                 # Historical data for validating our results - the number of primes
-                    100 : 25,               # to be found under some limit, such as 168 primes under 1000
-                    1000 : 168,
-                    10000 : 1229,
-                    100000 : 9592,
-                    1000000 : 78498,
-                    10000000 : 664579,
-                    100000000 : 5761455
-                  }
-
-    def __init__(this, limit):
-        this.sieveSize = limit
-        this.rawbits = [True] * (int((this.sieveSize+1)/2))
-
-    # Look up our count of primes in the historical data (if we have it) to see if it matches
-
-    def validateResults(this):                      # Check to see if this is an upper_limit we can
-        if this.sieveSize in this.primeCounts:      # the data, and (b) our count matches. Since it will return
-            return this.primeCounts[this.sieveSize] == this.countPrimes() # false for an unknown upper_limit, can't assume false == bad
+# Updated 7/7/2021 to be more pythonic/efficient
+
+from math import sqrt  # Used by the sieve
+import timeit  # For timing the durations
+
+
+PRIMECOUNTS = {
+    10: 4,  # Historical data for validating our results - the number of primes
+    100: 25,  # to be found under some limit, such as 168 primes under 1000
+    1000: 168,
+    10000: 1229,
+    100000: 9592,
+    1000000: 78498,
+    10000000: 664579,
+    100000000: 5761455,
+}
+
+
+class PrimeSieve(object):
```

Comment:
This capitalization pattern is more common in python. Also, `object` doesn't need to be explicit anymore, but I left it. could be removed.
```suggestion
class PrimeSieve:
```